mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,651 @@
+    tn, fp, fn, tp = confusion_matrix.ravel()
+    metrics["true_negatives"] = tn
+    metrics["false_positives"] = fp
+    metrics["false_negatives"] = fn
+    metrics["true_positives"] = tp
+    metrics["recall"] = sk_metrics.recall_score(y, y_pred)
+    metrics["precision"] = sk_metrics.precision_score(y, y_pred)
+    metrics["f1_score"] = sk_metrics.f1_score(y, y_pred)
+    return metrics
+
+
+def _get_classifier_global_metrics(is_binomial, y, y_pred, y_probs, labels):
+    """
+    get classifier metrics which computing over all classes examples.
+    """
+    metrics = {}
+    metrics["accuracy"] = sk_metrics.accuracy_score(y, y_pred)
+    metrics["example_count"] = len(y)
+
+    if not is_binomial:
+        metrics["f1_score_micro"] = sk_metrics.f1_score(y, y_pred, average="micro", labels=labels)
+        metrics["f1_score_macro"] = sk_metrics.f1_score(y, y_pred, average="macro", labels=labels)
+
+    if y_probs is not None:
+        metrics["log_loss"] = sk_metrics.log_loss(y, y_probs, labels=labels)
+
+    return metrics
+
+
+def _get_classifier_per_class_metrics_collection_df(y, y_pred, labels):
+    per_class_metrics_list = []
+    for positive_class_index, positive_class in enumerate(labels):
+        (y_bin, y_pred_bin, _,) = _get_binary_sum_up_label_pred_prob(
+            positive_class_index, positive_class, y, y_pred, None
+        )
+
+        per_class_metrics = {"positive_class": positive_class}
+        per_class_metrics.update(_get_classifier_per_class_metrics(y_bin, y_pred_bin))
+        per_class_metrics_list.append(per_class_metrics)
+
+    return pd.DataFrame(per_class_metrics_list)
+
+
+_Curve = namedtuple("_Curve", ["plot_fn", "plot_fn_args", "auc"])
+
+
+def _gen_classifier_curve(
+    is_binomial,
+    y,
+    y_probs,
+    labels,
+    curve_type,
+):
+    """
+    Generate precision-recall curve or ROC curve for classifier.
+    :param is_binomial: True if it is binary classifier otherwise False
+    :param y: True label values
+    :param y_probs: if binary classifer, the predicted probability for positive class.
+                    if multiclass classiifer, the predicted probabilities for all classes.
+    :param labels: The set of labels.
+    :param curve_type: "pr" or "roc"
+    :return: An instance of "_Curve" which includes attributes "plot_fn", "plot_fn_args", "auc".
+    """
+    if curve_type == "roc":
+
+        def gen_x_y_thresholds_fn(_y, _y_prob):
+            fpr, tpr, _thresholds = sk_metrics.roc_curve(_y, _y_prob)
+            return fpr, tpr, _thresholds
+
+        xlabel = "False Positive Rate"
+        ylabel = "True Positive Rate"
+        legend_loc = "lower right"
+    elif curve_type == "pr":
+
+        def gen_x_y_thresholds_fn(_y, _y_prob):
+            precision, recall, _thresholds = sk_metrics.precision_recall_curve(_y, _y_prob)
+            _thresholds = np.append(_thresholds, [1.0], axis=0)
+            return recall, precision, _thresholds
+
```

Comment:
We should calculate and append to the plot the Average Precision value as a series label. 
Alternatively, could we use the sklearn plots for this (it calculates and labels the plot for PR curves with the API value) instead (or if we're version restricted, could we emulate the plot style generated from this module)?
ref: https://scikit-learn.org/stable/auto_examples/model_selection/plot_precision_recall.html

Reply:
Similarly for ROC curve, could we display the AUC value on the plots?
similar to how sklearn 1.0 does it: https://scikit-learn.org/stable/modules/generated/sklearn.metrics.RocCurveDisplay.html#sklearn.metrics.RocCurveDisplay.from_predictions

Reply:
Question: shall we also display Average Precision value and AUC value on multi-class plots ? (each line has its own 2 values)